src/main/java/com/here/platform/artifact/maven/wagon/ArtifactWagon.java

```diff
@@ -276,9 +277,30 @@ public void put(File source, String resourceName)
       super.put(source, resourceName);
     } catch (AuthorizationException e) {
       throw new AuthorizationException(AUTHORIZATION_FORBIDDEN_ERROR_MESSAGE, e);
+    } catch (RuntimeException re){
+      throw new TransferFailedException(String.format(FILE_PUT_ERROR_MESSAGE, resourceName), re);
     }
   }
 
+  @Override
+  protected CloseableHttpResponse execute(HttpUriRequest httpMethod) throws HttpException, IOException {
+    CloseableHttpResponse httpResponse = super.execute(httpMethod);
+
+      int status = httpResponse.getStatusLine().getStatusCode();
+      if (status == HttpStatus.SC_UNPROCESSABLE_ENTITY) {
```

Comment:
Could we handle other possible errors here? I mean change condition to `status >= 400`
This might be helpful because Artifact Service returns descriptive errors descriptions